### PR TITLE
Use Webpack to improve extension performance

### PR DESCRIPTION
**Pull request by LazarusX**

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -7,22 +7,29 @@
...
-            "sourceMaps": true,
-            "outFiles": [ "${workspaceRoot}/out/src/**/*.js" ],
-            "preLaunchTask": "npm"
+            "args": [
+                "--extensionDevelopmentPath=${workspaceFolder}"
+            ],
+            "outFiles": [
+                "${workspaceFolder}/dist/**/*.js"
+            ],
+            "preLaunchTask": "npm: webpack"
```
Comment (adashen) [inline .vscode/launch.json]: webpack or webpack-dev?

File: .vscode/launch.json [hunk 2]
```diff
@@ -7,22 +7,29 @@
...
+                "--extensionTestsPath=${workspaceRoot}/out/test"
+            ],
             "stopOnEntry": false,
             "sourceMaps": true,
-            "outFiles": [ "${workspaceRoot}/out/test/**/*.js" ],
-            "preLaunchTask": "npm"
+            "outFiles": [
+                "${workspaceRoot}/out/test/**/*.js"
+            ],
+            "preLaunchTask": "npm: compile"
```
Comment (adashen) [inline .vscode/launch.json]: why remove the npm task? It has some problem watch check.

File: package.json [hunk 3]
```diff
@@ -411,11 +411,13 @@
     ]
   },
   "scripts": {
-    "vscode:prepublish": "tsc -p ./",
-    "compile": "tsc -watch -p ./",
+    "vscode:prepublish": "webpack --mode production",
+    "compile": "tsc -p ./",
```
Comment (adashen) [inline package.json]: why remote watch mode?

File: .vscode/launch.json (hunk 2 above)
Comment (LazarusX) [reply .vscode/launch.json]: VS Code reports schema deprecation warning of current task.json. After removing task.json, VS Code can automatically discovery npm scripts as tasks and we don't need to configure them manually anymore.

File: .vscode/launch.json (hunk 1 above)
Comment (LazarusX) [reply .vscode/launch.json]: `webpack-dev` will never return so the extension host cannot start.

File: package.json (hunk 3 above)
Comment (adashen) [reply package.json]: it is used in "launch tests"?

File: .vscode/launch.json (hunk 1 above)
Comment (adashen) [reply .vscode/launch.json]: so what is webpack-dev used for?

Comment (LazarusX) [reply .vscode/launch.json]: It's copied from the sample, and not used currently.

File: package.json (hunk 3 above)
Comment (LazarusX) [reply package.json]: Yes, that's also the use case of CI.

Comment (adashen) [reply package.json]: So now the test is not compiled in watch mode. Any impact? Will the reload still work?

Comment (LazarusX) [reply package.json]: No. We seldom run the test launch profile locally in watch mode in VS Code, as VS Code does not allow running the test with any VS Code instance open.

**Pull request merged**